src/explorer.ts

```diff
@@ -91,4 +111,39 @@ class KubernetesResource implements KubernetesObject, ResourceNode {
     constructor(kind: kuberesources.ResourceKind, readonly id: string) {
         this.resourceId = kind.abbreviation + '/' + id;
     }
-}
\ No newline at end of file
+}
+
+class VirtualKubernetesResource implements KubernetesObject, ResourceNode {
+    readonly resourceId: string;
+
+    constructor (readonly id : string, readonly kind : string, readonly metadata?: any) {
+        this.resourceId = kind + "/" + id;
+    }
+
+    getChildren() : vscode.ProviderResult<KubernetesObject[]> {
+        if (this.kind === 'cluster') {
```

Comment:
Presumably this will never be called for inactive clusters because of the `TreeItemCollapsibleState.None`?  We don't (yet!) want to display resources for inactive clusters because that will require a more substantial rewriting of the infrastructure (which is not a bad thing, just a lot of extra effort!).

Reply:
Correct, for inactive clusters, the `getChildren()` will never be called. Just show their names as placeholder, not expandable at the tree explorer.